icicle-compiler/test/Icicle/Test/Arbitrary/Corpus.hs

```diff
@@ -50,94 +49,154 @@ inputs =
-   , "           | True -> (map_insert location severity (fst x), Some time)"
-   , "           | False -> (fst x, Some time)"
-   , "           end"
-   , "         | False ->"
-   , "           case is_homer"
-   , "           | True -> (map_insert location severity map_create, Some time)"
-   , "           | False -> (map_create, Some time)"
-   , "           end"
-   , "         end"
-   , "       end"
-   , "     end"
-   , "~> (group fold (k, v) = fst x ~> min v)"
-   ])
- ]
-
-
-
-wellTypedCorpus :: Gen WellTyped
-wellTypedCorpus = genCore >>= genWellTyped
-
-testAllCorpus :: (WellTyped -> Property) -> Property
-testAllCorpus prop
- = conjoin $ fmap run queries
- where
-  run q = forAll (gen q) prop
-
-  gen (name,src)
-   = case coreOfSource name src of
-      Left u -> Savage.error (Text.unpack (renderOutputId name) <> ":\n" <> Text.unpack src <> "\n\n" <> u)
-      Right core -> genWellTyped (name, core)
-
-genWellTyped :: (OutputId, C.Program () Var) -> Gen WellTyped
-genWellTyped (name, core)
- = do wt <- tryGenWellTypedFromCoreEither S.AllowDupTime (streamType core) core
-      case wt of
-       Left err  -> Savage.error (Text.unpack (renderOutputId name) <> "\n\n" <> err)
-       Right wt' -> return wt'
- where
-  streamType c
-   = InputType $ C.inputType c
-
-
-genCore :: Gen (OutputId, C.Program () Var)
-genCore = do
-  (name,src) <- elements queries
-  case coreOfSource name src of
-   Left u -> Savage.error (Text.unpack (renderOutputId name) <> ":\n" <> Text.unpack src <> "\n\n" <> u)
-   Right core -> return (name, core)
+corpusInputId :: InputName -> InputId
+corpusInputId = InputId [namespace|corpus|]
+
+corpusQueries :: [(InputName, OutputId, Text)]
+corpusQueries =
+  [
+    -- Tombstones/no values treament between Core and C
+
+    ( [inputname|int|]
+    , [outputid|tombstone:bool|]
+    , "feature int ~> fold x = False : True ~> x")
+
+  , ( [inputname|injury|]
+    , [outputid|tombstone:group|]
+    , "feature injury ~> group location ~> count location ")
+
+  , ( [inputname|injury|]
+    , [outputid|tombstone:group_group|]
+    , "feature injury ~> group location ~> group location ~> count location ")
+
+  , ( [inputname|injury|]
+    , [outputid|tombstone:group_group_group|]
+    , "feature injury ~> group location ~> group location ~> group location ~> count location ")
+
+  , ( [inputname|int|]
+    , [outputid|tombstone:filter|]
+    , "feature int ~> filter tombstone ~> count value")
+
+  , ( [inputname|injury|]
+    , [outputid|tombstone:filter_group|]
+    , "feature injury ~> filter tombstone ~> group location ~> count location ")
+
+  , ( [inputname|injury|]
+    , [outputid|tombstone:filter_group_group|]
+    , "feature injury ~> filter tombstone ~> group location ~> group location ~> count location ")
+
+  , ( [inputname|injury|]
+    , [outputid|tombstone:filter_group_group_group|]
+    , "feature injury ~> filter tombstone ~> group location ~> group location ~> group location ~> count location ")
```

Comment:
just focusing on tombstones for these queries, there are groups on different keys below this